Library/src/main/java/com/shopify/testify/internal/compare/RegionCompare.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Shopify Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.shopify.testify.internal.compare
+
+import android.graphics.Bitmap
+import android.graphics.Rect
+import androidx.annotation.ColorInt
+
+internal class RegionCompare(private val exclusionRects: Set<Rect>) : BitmapCompare {
+
+    override fun compareBitmaps(baselineBitmap: Bitmap, currentBitmap: Bitmap): Boolean {
+        if (baselineBitmap.height != currentBitmap.height) {
+            return false
+        }
+
+        if (baselineBitmap.width != currentBitmap.width) {
+            return false
+        }
+
+        for (y in 0 until baselineBitmap.height) {
```

Comment:
This is probably fine for now but I think it would be worth looking into a faster method

Reply:
I've looked at this before and the real crappy part is the `getPixel` methods on `Bitmap`. They're super slow. I'm not sure if there's a faster way to just load the entire bitmap into memory either. The bitmaps are encoded with different color spaces and alpha arrangement (pre-multiplied alpha sometimes) and the `getPixel` code resolves that for us. I would much prefer a `memcmp` style, but I haven't invested enough in figuring out how hard that is.

Was there a faster method you'd expect to see being used here? Maybe I'm just thinking about this the wrong way

Reply:
I was expecting to see something with `setPixel` but I don't know what the implementations of the `Bitmap` methods look like, so maybe that isn't actually all that fast.